Adding matched track quality variables for pfcandAnalyzer

abaty opened the pull request:
Describe the Pull-Request:

This change adds an option to loop over the track collection and match tracks with their charged pf candidate partner in the pfcandAnalyzer tree.  The quality variables of the matched track are then output into the tree.  5 branches have been added.  The matching between pfcand and track should be exact (not done via eta/phi matching).

Does the PR pass the test script located at
HeavyIonsAnalysis/JetAnalysis/test/runtest.sh
[X] Yes
[] No

Please make sure to mention (using the @ syntax) anyone who might be interested in this PR.
@cfmcginn Could you please take a look at the distributions and make sure the matching looks reasonable to you?

Commit:
added matched track quality info to pfcand tree for debugging

Commit:
fix to config

bi-ran commented on HeavyIonsAnalysis/JetAnalysis/src/HiPFCandAnalyzer.cc:
```diff
@@ -85,6 +99,49 @@ HiPFCandAnalyzer::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetu
+      unsigned int trackKey = pfcand.trackRef().key();     
+
+      //loop through tracks looking for this key 
+      for(unsigned int it=0; it<tracks->size(); ++it){
```
is the loop needed if the key is exactly the index?

bi-ran commented:
thanks! @abaty 

there are no matched tracks for the cs pf candidates. can you please confirm that this is expected, and disable it for the pfcandAnalyzerCS tree if so?

mverwe commented:
if we want track level info for CS PF candidates, you have to first go back to the original PF candidate using sourceCandidatePtr(0). However, the problem should be really solved prior to running the subtraction.

Commit:
removed track matching from pfCandCS analyzer, simplified track key lookup

Commit:
changed maxEvent back

abaty commented:
I've removed the track matching from the CS analyzer.  Also simplified the track key lookup to not require a loop.

One thing to note: charged hadrons and muons seem to have good track matching, but I am not sure if the track matching will work appropriately for electrons b/c they use a different track collection.  It would be good if an electron expert can investigate this, but I think that should not hold up this PR.

bi-ran commented:
the gsf track collection is separate from the general tracks collection, but it is possible that the same seed is used for both, which will give the electron object a reference to the track in the general track collection. in general, there is no guarantee of a match existing.

@cfmcginn do you have any comments?

abaty commented:
Hi,

Does anyone have any other objections to merging this?

@cfmcginn @bi-ran

abaty commented:
Friendly reminder that this is still not merged.  @bi-ran

bi-ran commented:
@abaty sorry for the delay
